impeller/entity/contents/text_contents.cc

```diff
@@ -61,33 +55,19 @@ std::optional<Rect> TextContents::GetCoverage(const Entity& entity) const {
   return bounds->TransformBounds(entity.GetTransformation());
 }
 
-bool TextContents::Render(const ContentContext& renderer,
-                          const Entity& entity,
-                          RenderPass& pass) const {
-  if (color_.IsTransparent()) {
-    return true;
-  }
-
-  auto atlas = ResolveAtlas(renderer.GetContext());
-
-  if (!atlas || !atlas->IsValid()) {
-    VALIDATION_LOG << "Cannot render glyphs without prepared atlas.";
-    return false;
-  }
-
-  using VS = GlyphAtlasPipeline::VertexShader;
-  using FS = GlyphAtlasPipeline::FragmentShader;
-
-  // Information shared by all glyph draw calls.
-  Command cmd;
-  cmd.label = "TextFrame";
-  cmd.primitive_type = PrimitiveType::kTriangle;
-  cmd.pipeline =
-      renderer.GetGlyphAtlasPipeline(OptionsFromPassAndEntity(pass, entity));
-  cmd.stencil_reference = entity.GetStencilDepth();
+template <class VS_, class FS_>
```

Comment:
The template type param can just be the pipeline type itself rather than its shaders (see the advanced blends in BlendFilter for an example of this).

Reply:
Nice, doen